src/condor_utils/dagman_utils.cpp

```diff
@@ -447,11 +447,13 @@ DagmanUtils::setUpOptions(DagmanOptions &options, str_list &dagFileAttrLines)
 	if (options[deep::str::DagmanPath].empty()) {
 		fprintf(stderr, "ERROR: can't find %s in PATH, aborting.\n",
 		        dagman_exe );
+		if (err) { *err = std::string("Failed to locate ") + dagman_exe + " executable."; }
```

Comment:
Stylistically, I think this would be better-phrased as constructing (via `format_str()`) an error message, calling `fprintf(stderr, "ERROR: %s\n", error_message.c_str();` and then doing the conditional assignment to `* err`, unless there's some good reason for the error message to be inconsistent.

Whether this function should be trying to print to stderr, given that's it's called by the Python bindings, is another question and probably out of scope for this PR.